src/plugins/intel_cpu/src/nodes/kernels/x64/rope_kernel.cpp

```diff
@@ -89,10 +89,16 @@ void jit_rotary_kernel<isa>::rotary_half(size_t step) {
     vfmsub231ps(vmm_dst0, vmm_cos, vmm_src0);
     store(reg_dst, vmm_dst0, m_jcp.dst_prc, step);
 
-    // cos[i + halfRotaryNdims]
-    load(vmm_cos, reg_cos, ov::element::f32, step, false, half_rotary_ndims * sizeof(float));
-    // sin[i + halfRotaryNdims]
-    load(vmm_sin, reg_sin, ov::element::f32, step, false, half_rotary_ndims * sizeof(float));
+    // cos[i + sin_cos_offset]
+    // if con/sin table is not same size with input, it is reused for both halves.
+    const size_t sin_cos_offset = m_jcp.cos_sin_ndims == half_rotary_ndims ? 0 : half_rotary_ndims;
+    if (sin_cos_offset) {
+        load(vmm_cos, reg_cos, ov::element::f32, step, false, half_rotary_ndims * sizeof(float));
+    }
+    // sin[i + sin_cos_offset]
+    if (sin_cos_offset) {
+        load(vmm_sin, reg_sin, ov::element::f32, step, false, half_rotary_ndims * sizeof(float));
+    }
```

Comment:
`sin_cos_offset` is used as a `bool` flag. Isn't it more logical to use bool type?